pkg/db/db.go

```diff
@@ -307,7 +308,7 @@ const (
 // ESL EVENTS
 
 // DBWriteEslEventInternal writes one event to the event-sourcing-light table, taking arbitrary data as input
-func (h *DBHandler) DBWriteEslEventInternal(ctx context.Context, eventType EventType, tx *sql.Tx, data interface{}) error {
+func (h *DBHandler) DBWriteEslEventInternal(ctx context.Context, eventType EventType, tx *sql.Tx, data interface{}, metadata interface{}) error {
```

Comment:
the `metadata` should have a concrete type (a struct), and not the generic interface